signing/sawtooth_signing/core.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2016, 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+from abc import ABCMeta
+from abc import abstractmethod
+
+
+class NoSuchAlgorithmError(Exception):
+    """Thrown when trying to create an algorithm which does not exist.
+    """
+    pass
+
+
+class SigningError(Exception):
+    """Thrown when an error occurs during the signing process.
+    """
+    pass
+
+
+class ParseError(Exception):
+    """Thrown when an error occurs during deserialization of a Private or
+    Public key from various formats.
+    """
+    pass
+
+
+class PrivateKey(metaclass=ABCMeta):
+    """A private key instance.
```

Comment:
Any reason not to add new_random() to this ABC?

Reply:
new_random() doesn't make sense on an instance; however, maybe it would make sense on context - context.new_random_private_key().

Reply:
what about just creating a new PrivateKey(). You don't need a separate member method. A create without parameters should call a private new random method.

Reply:
I'll add `new_random_private_key` to context.  It should remove many imports of `Secpk2561PrivateKey`